dotCMS/src/integration-test/java/com/dotcms/security/secret/ServiceIntegrationAPIImplTest.java

```diff
@@ -0,0 +1,344 @@
+                .build();
+
+        api.saveSecrets(beanSystemHost1, systemHost, admin);
+
+        final Optional<ServiceSecrets> serviceSecretsOptional = api.getSecrets("serviceKey-1-Any-Host", true, host1, admin);
+        assertTrue(serviceSecretsOptional.isPresent());
+        final ServiceSecrets recoveredBean = serviceSecretsOptional.get();
+        assertEquals("serviceKey-1-Any-Host", recoveredBean.getServiceKey());
+        assertTrue(recoveredBean.getSecrets().containsKey("fallback:test"));
+        assertTrue(recoveredBean.getSecrets().containsKey("fallback:test:secret1"));
+    }
+
+    @Test
+    public void Test_Save_Secrets_Then_Replace_Secret() throws DotDataException, DotSecurityException {
+        final ServiceIntegrationAPI api = APILocator.getServiceIntegrationAPI();
+        final User admin = TestUserUtils.getAdminUser();
+        final Host host = new SiteDataGen().nextPersisted();
+
+        final ServiceSecrets.Builder builder1 = new ServiceSecrets.Builder();
+        final ServiceSecrets secrets1 = builder1.withServiceKey("serviceKey-1-Host-1")
+                .withHiddenSecret("test:secret1", "sec1")
+                .withHiddenSecret("test:secret2", "sec2")
+                .build();
+        api.saveSecrets(secrets1, host, admin);
+
+        final ServiceSecrets.Builder builder2 = new ServiceSecrets.Builder();
+        final ServiceSecrets secrets2 = builder2.withServiceKey("serviceKey-1-Host-1")
+                .withHiddenSecret("test:secret1", "secret1")
+                .build();
+        api.saveSecrets(secrets2, host, admin);
+        final Optional<ServiceSecrets> serviceSecretsOptional = api.getSecrets("serviceKey-1-Host-1", host, admin);
+        assertTrue(serviceSecretsOptional.isPresent());
+        final ServiceSecrets recoveredBean = serviceSecretsOptional.get();
+        assertEquals("serviceKey-1-Host-1", recoveredBean.getServiceKey());
+        assertTrue(recoveredBean.getSecrets().containsKey("test:secret1"));
+        assertFalse(recoveredBean.getSecrets().containsKey("test:secret2"));
+        assertEquals("secret1", recoveredBean.getSecrets().get("test:secret1").getString());
+    }
+
+    @Test
+    public void Test_Save_Secrets_Then_Update_Single_Secret_Verify_Updated_Value() throws DotDataException, DotSecurityException {
+        final ServiceIntegrationAPI api = APILocator.getServiceIntegrationAPI();
+        final User admin = TestUserUtils.getAdminUser();
+        final Host host = new SiteDataGen().nextPersisted();
+
+        //Let's create a set of secrets for a service
+        final ServiceSecrets.Builder builder1 = new ServiceSecrets.Builder();
+        final ServiceSecrets secrets1 = builder1.withServiceKey("serviceKey-1-Host-1")
+                .withHiddenSecret("test:secret1", "secret-1")
+                .withHiddenSecret("test:secret2", "secret-2")
+                .withHiddenSecret("test:secret3", "secret3")
+                .withHiddenSecret("test:secret4", "secret-4")
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String literal "test:secret4" appears 8 times in this file; the first occurrence is on line 203](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4623512)